asab/application.py

```diff
@@ -194,6 +199,28 @@ def handler(type):
 		# Every 10 minutes listen for housekeeping
 		self.PubSub.subscribe("Application.tick/600!", self._on_housekeeping_tick)
 
+		# Run the watchdog to detect lost interactivity on the event loop
+		watchdog_thread = threading.Thread(target=self._watchdog)
+		watchdog_thread.daemon = True  # Daemonize the thread to ensure it exits with the main program
+		watchdog_thread.start()
+
+
+	def _watchdog(self):
+		"""
+		Periodically checks the loop interactivity.
```

Comment:
... and maily KILLing a stuck process by a signal - that's an important bit to say